### ISSUE TITLE
[Bug]: expo prebuild fails in a monorepo.

### ISSUE THREAD

**Issue by phi-tz** [OneSignal/onesignal-expo-plugin#81]
### What happened?

My codebase is in a monorepo with the following structure:

```
root
-> mobile
    -> package.json
    -> app.config.js
```

Running `expo prebuild` fails with the error:
```
onesignal-expo-plugin: Couldn't read file:node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/NotificationService.h
```

it is fixed if I dont hoist `onesignal-expo-plugin`, i.e. by adding the following to the root's `package.json`:
```
 "nohoist": [
    "**/onesignal-expo-plugin"
]
```

### Steps to reproduce?

```Markdown
As described above.
```

### What did you expect to happen?

Running `expo prebuild` in a monorepo should work.

### OneSignal Expo SDK version

1.0.1

### Platform

iOS

### Relevant log output

```Shell
Created native projects | gitignore already synced
⚠️  Skipped Metro config updates:
› Existing Metro config found; not overwriting.
› You will need to extend the default @expo/metro-config in your Metro config.
  Learn more: https://docs.expo.dev/guides/customizing-metro

✔ Updated package.json and added index.js entry point for iOS and Android
🧶 Using Yarn to install packages. Pass --npm to use npm instead.
✔ Installed JavaScript dependencies 3253ms
» android: userInterfaceStyle: Install expo-system-ui in your project to enable this feature.
Using node to generate images. This is much slower than using native packages.
› Optionally you can stop the process and try again after successfully running `npm install -g sharp-cli`.

Using node to generate images. This is much slower than using native packages.
› Optionally you can stop the process and try again after successfully running `npm install -g sharp-cli`.

        onesignal-expo-plugin: Couldn't read file:node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/NotificationService.h
(node:38807) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/NotificationService.h'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:38807) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:38807) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✔ Config synced
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: Unable to find a target named `OneSignalNotificationServiceExtension` in project `TradeziDev.xcodeproj`, did find `TradeziDev`.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by rgomezp** [comment]
Howdy,
Thanks for reporting. We just released version 1.0.1. If you revert to 1.0.0 do you see the issue still?

**Comment by phi-tz** [comment]
Yes I'm seeing the same issue:

```
 Created native projects | /android, /ios already created | gitignore already synced
⚠️  Skipped Metro config updates:
› Existing Metro config found; not overwriting.
› You will need to extend the default @expo/metro-config in your Metro config.
  Learn more: https://docs.expo.dev/guides/customizing-metro

✔ Updated package.json and added index.js entry point for iOS and Android
🧶 Using Yarn to install packages. Pass --npm to use npm instead.
✔ Installed JavaScript dependencies 1434ms
        onesignal-expo-plugin: OneSignal dependencies already added to build.gradle. Skipping...
» android: userInterfaceStyle: Install expo-system-ui in your project to enable this feature.
        onesignal-expo-plugin: OneSignalNotificationServiceExtension target already added to Podfile. Skipping...
        onesignal-expo-plugin: Couldn't read file:node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/NotificationService.h
(node:48301) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open 'node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/NotificationService.h'
(Use `node --trace-warnings ...` to show where the warning was created)
(node:48301) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:48301) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
✔ Config synced
```

**Comment by rgomezp** [comment]
In your hoisted `node_modules`, you should see the NSE support files under `node_modules/onesignal-expo-plugin/build/support/serviceExtensionFiles/` including `NotificationService.h`. Take a look in there to see that the files are there.

**Comment by phi-tz** [comment]
I've been using monorepo from the start.  `expo prebuild` never worked.  I could only get it to work by unhoisting the package as a workaround.  This issue is for adding support for monorepo, which is the case for a lot of developers.

**Comment by rgomezp** [comment]
I'm not quite sure what the file structure looks like. Perhaps you can help me so that we can get it fixed.

The issue stems from the fact that some Javascript in the module uses a relative file path. So the question is, how does the structure change at the `node_module` level?

Also, if you could provide any steps you took to convert into a monorepo, that might be helpful in case we need to reproduce.

**Comment by brunolcarlos** [comment]
I'm getting a similar error

Could not load RNOneSignal native module. Make sure native dependencies are properly linked.
at node_modules/react-native-onesignal/dist/helpers.js:15:8 in isNativeModuleLoaded
at node_modules/react-native-onesignal/dist/index.js:61:13 in OneSignal.setAppId
at App.js:13:4 in useEffect$argument_0
at node_modules/react-native/Libraries/ReactNative/renderApplication.js:58:4 in renderApplication
at node_modules/react-native/Libraries/ReactNative/AppRegistry.js:117:25 in runnables.appKey.run
at node_modules/react-native/Libraries/ReactNative/AppRegistry.js:202:4 in runApplication

`"dependencies": {
    "@react-native-async-storage/async-storage": "~1.15.0",
    "expo": "~43.0.0",
    "expo-constants": "~12.1.3",
    "expo-device": "~4.0.3",
    "expo-file-system": "~13.0.3",
    "expo-media-library": "~13.0.3",
    "expo-splash-screen": "~0.13.5",
    "expo-status-bar": "~1.1.0",
    "expo-system-ui": "~1.0.0",
    "expo-updates": "~0.10.13",
    "expo-web-browser": "~10.0.3",
    "onesignal-expo-plugin": "^1.0.0 ",
    "react": "17.0.1",
    "react-dom": "^18.2.0",
    "react-native": "0.64.3",
    "react-native-device-info": "^9.0.2",
    "react-native-gesture-handler": "~1.10.2",
    "react-native-navigation-bar-color": "^2.0.1",
    "react-native-onesignal": "^4.3.11",
    "react-native-reanimated": "~2.2.0",
    "react-native-safe-area-context": "3.3.2",
    "react-native-screens": "~3.8.0",
    "react-native-unimodules": "~0.15.0",
    "react-native-web": "^0.18.1",
    "react-native-webview": "11.13.0"
  },`

**Comment by mauricedoepke** [comment]
I got the same error in a pnpm monorepo

**Comment by brismithers** [comment]
Hi!

We've just published version 1.1.0 of the plugin which includes a fix to support monorepo, please let me know if you have any issues.  Thank you for your patience!

**Comment by crieggalder** [comment]
Still running into issues - in our case (Expo + Lerna) we have a node_modules dir in our app dir (so at same level as Expo ios/android dirs) and another node_modules dir above the app dir. It seems like the plugin is just finding node_modules in the app dir, but the onesignal dependencies are being hoisted all the way up to the top dir.

Can we tweak this to keep looking in parent dirs for additional node_modules dirs if the first found doesn't contain deps?
